### DISCUSSION TITLE
Questions about capabilities

### DISCUSSION THREAD

**Discussion by Dirt-m** [Aterfax/DCS-World-Dedicated-Server-Docker#81] [Q&A] [answered]
Hey! I'm a DCS server owner looking to switch my server needs over to docker, i spent some time looking into docker, but i'm not sure if i quite understand the capabilities of this docker image. I'm quite new to docker so please bear with me and my basic questions:

Can i run multiple dcs servers in one container? the reason for this question is that i want my DCSSB to be connected to two servers simultaneously, if there is a different way to do this then please let me know.

How does mission uploading work, can i upload them on the fly and do my missions persist after a server restart?

Do i need a GUI to use the server?

Thats all (for now :p), thanks!

**Comment by Aterfax** [comment, marked as answer]
**>Can i run multiple dcs servers in one container?**

Yes, but that is not currently how the native docker image is opinionated. If you wanted to do this I would recommend using the DCSSB mod and using the native functionality of DCSSB.

I don't know enough about the functionality of DCSSB to suggest how you would connect it to two servers not on the same machine. I would run separate instances of DCSSB for any DCS instance not running within the same container/machine instance.

-----

**>How does mission uploading work, can i upload them on the fly**

If the query is on how to use the docker container, then you can upload files using the KASM (side) menu's file manager. You would then either upload them to the correct place up front, or then shift them to the correct location with the GUI file manager.

-----

**>do my missions persist after a server restart?**

If you mean do mission files that have been uploaded get deleted, then no. They do not,

If you mean in terms of a the running mission environment, without wanting to be too terse: if you don't know then the answer will be no. DCS missions need to be purpose built to have this kind of "history" persistence. 

-----

**>Do i need a GUI to use the server?**

This depends on what scope you are talking about. On the host running the docker container, no. The container itself starts a virtual GUI which you access from the web browser, thus from the machine you want to administrate the running server, yes.

-----

As a general aside, if you have Q's around the running of "a DCS server" rather than running one with this docker specifically, I'm not likely going to answer as it doesn't really lie within the scope of this project.

If you did want some general help, with DCS server or DCSSB itself, then the DCSSB Discord server will probably be a good shout:

https://github.com/Special-K-s-Flightsim-Bots/DCSServerBot?tab=readme-ov-file#contact--support

**Reply by Dirt-m** [reply]
Thanks for your reply! i think i got everything i need
